BSP/sharpMemoryLCD/LS013B7DH03.c

```diff
@@ -0,0 +1,206 @@
+
+    GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+    GPIO_InitStruct.Pin = GPIO_PIN_0;
+    /* Pin D0 is SPI2 NSS */
+    /* It is not driven by SPI2 peripheral because need to be enabled with high level */
+    HAL_GPIO_Init(GPIOD, &GPIO_InitStruct);
+    LCDslaveSelect(false);
+
+    GPIO_InitStruct.Pin = GPIO_PIN_6;
+    /* Pin D6 is DISP (for enabling display) */
+    HAL_GPIO_Init(GPIOD, &GPIO_InitStruct);
+    LCDdisplayEnable(false);
+
+
+    /* Output a 60kHz signal to EXTCOMIN pin */
+    TIM_HandleTypeDef TimHandle;
+    TimHandle.Instance = TIM3;
+    TimHandle.Init.Period        = 65535;
+    TimHandle.Init.Prescaler     = 60000; // counter is counting at 120M/60k => 2kHz
+    TimHandle.Init.ClockDivision = TIM_CLOCKDIVISION_DIV4; // 500Hz
+    TimHandle.Init.CounterMode   = TIM_COUNTERMODE_UP;
+    TimHandle.Init.Period        = 8; //500/8 => 60Hz period
+
+    if (HAL_TIM_PWM_Init(&TimHandle) != HAL_OK)
+    {
+        assert_param(false);
+    }
+
+    TIM_OC_InitTypeDef sConfig;
+    sConfig.OCMode       = TIM_OCMODE_PWM1;
+    sConfig.OCPolarity   = TIM_OCPOLARITY_HIGH;
+    sConfig.OCFastMode   = TIM_OCFAST_DISABLE;
+    sConfig.OCNPolarity  = TIM_OCNPOLARITY_HIGH;
+    sConfig.OCNIdleState = TIM_OCNIDLESTATE_RESET;
+    sConfig.OCIdleState  = TIM_OCIDLESTATE_RESET;
+    sConfig.Pulse = 1;
+    if (HAL_TIM_PWM_ConfigChannel(&TimHandle, &sConfig, TIM_CHANNEL_1) != HAL_OK)
+    {
+        assert_param(false);
+    }
+
+    if (HAL_TIM_PWM_Start(&TimHandle, TIM_CHANNEL_1) != HAL_OK)
+    {
+        assert_param(false);
+    }
+
+    SpiHandle.Instance               = SPI2;
+    SpiHandle.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_256; // ===> TODO set correct baudrate to 1Mbps
+    SpiHandle.Init.Direction         = SPI_DIRECTION_1LINE;
+    SpiHandle.Init.CLKPhase          = SPI_PHASE_1EDGE;
+    SpiHandle.Init.CLKPolarity       = SPI_POLARITY_LOW;
+    SpiHandle.Init.DataSize          = SPI_DATASIZE_8BIT;
+    SpiHandle.Init.FirstBit          = SPI_FIRSTBIT_LSB;
+    SpiHandle.Init.TIMode            = SPI_TIMODE_DISABLE;
+    SpiHandle.Init.CRCCalculation    = SPI_CRCCALCULATION_DISABLE;
+    SpiHandle.Init.NSS               = SPI_NSS_SOFT;
+    SpiHandle.Init.NSSPMode          = SPI_NSS_PULSE_DISABLE;
+    SpiHandle.Init.Mode              = SPI_MODE_MASTER;
+    if(HAL_SPI_Init(&SpiHandle) != HAL_OK)
+    {
+        assert_param(false);
+    }
+
+    sharpMemoryLCD_clearScreen();
+
+    LCDdisplayEnable(true);
+}
+
+
+void sharpMemoryLCD_clearScreen(void)
+{
+    /* To clear sreen send CMD_CLEAR (3bit) + 16 additional dummy bits */
+    uint8_t buffer[] = {CMD_CLEAR, 0x00};
+    LCDslaveSelect(true);
+    if(HAL_SPI_Transmit(&SpiHandle, buffer, sizeof(buffer), 1000) != HAL_OK)
+    {
+        assert_param(false);
+    }
+    LCDslaveSelect(false);
+}
+
+
+void sharpMemoryLCD_printTextLine(uint8_t line, const char *text, uint8_t nbChar)
+{
+    uint8_t buffer[Font16.Height*SCREEN_LENGTH/Font16.Width];
```

Comment:
Introduce a constant, or at least a variable for this monster :D
Also the var name can be enhanced : I know it is a buffer at least, but I want more please